Axios로 label api 호출해 label list 받아오기

pieisland opened the pull request:
- axios와 useEffect를 사용해 데이터베이스에 들어있는 label 값들을 받아왔습니다.
- 이슈 리스트 페이지의 LabelFilterButton과 이슈 상세 페이지의 LabelSelector에 적용했습니다.
- 지금은 useState로 label 상태를 관리하기 때문에 후에 useContext를 사용할 때 수정이 필요해 보입니다.

pieisland committed:
[Add] Get Labels Data

Get Labels data using axios.

pieisland committed:
[Add] Get Labels Data

Get Labels data using axios.

gyim1345 commented on front/src/components/LabelsSelector.js:
```diff
@@ -164,9 +166,35 @@ const tempData = [
+      return data;
+    })
+    .catch((err) => res.status(500).json({ message: err.message }));
+
```
서버와 비슷하게 async/await으로 통일하면 좋을것 같습니다!

gyim1345 reviewed (changes requested):
api 요청 async await으로 수정 부탁드립니다!

profornnan commented on front/src/components/LabelFilterButton.js:
```diff
@@ -168,17 +170,51 @@ const tempLabel = [
+    .catch((err) => res.status(500).json({ message: err.message }));
+
+  return db_data;
+};
```
저도 이슈 목록 페이지에서 API 요청하는 부분을 구현하다가 든 생각인데 API 요청 함수를 따로 디렉터리를 만들어서 관리해도 좋을 것 같아요!!

pieisland committed:
[Refactor] Label API

Use async await with try catch instead of promise then.
Add alert when calling api is falied.
Chnage variable name.

gyim1345 reviewed (approved):
수고하셨습니다!

rockpell reviewed (approved):
수고하셨습니다!

profornnan reviewed (approved):
고생하셨습니다!!
